plover/formatting.py

```diff
@@ -698,23 +700,25 @@ def _atom_to_action_spaces_after(atom, last_action):
             if (meta.startswith(META_RETRO_FORMAT) and meta.endswith(')')):
                 dict_format = meta[len(META_RETRO_FORMAT):-len(')')]
                 action = last_action.copy_state()
-                action.replace = last_action.word + SPACE
+                cast_input = False
                 try:
-                    float(last_action.word)
+                    float_word = float(last_action.word)
                 except ValueError:
                     pass
                 else:
-                    format = dict_format.replace('c', '{:,.2f}')
-                    cast_input = float(last_action.word)
+                    currency_format = dict_format.replace('c', '{:,.2f}')
+                    cast_input = float_word
                 try:
-                    int(last_action.word)
+                    integer_word = int(last_action.word)
                 except ValueError:
                     pass
                 else:
-                    format = dict_format.replace('c', '{:,}')
-                    cast_input = int(last_action.word)
-                action.text = format.format(cast_input) + SPACE
-                action.word = format.format(cast_input)
+                    currency_format = dict_format.replace('c', '{:,}')
+                    cast_input = integer_word
+                if cast_input is not False:
+                    action.replace = last_action.word + SPACE
+                    action.word = currency_format.format(cast_input)
+                    action.text = action.word + SPACE
         elif meta.startswith(META_COMMAND):
```

Comment:
Same as above. How about factorizing the code a little? The only difference is the extra space:
```diff
                 if cast_input is not None:
-                    action.replace = last_action.word
+                    action.replace = last_action.word + SPACE
                     action.word = currency_format.format(cast_input)
-                    action.text = action.word
+                    action.text = action.word + SPACE
```